OpenRA.Mods.Common/AI/SupportPowerDecision.cs

```diff
@@ -111,13 +141,57 @@ public int GetAttractiveness(IEnumerable<FrozenActor> frozenActors, Player fired
 			return answer;
 		}
 
+		/// <summary>Evaluates the attractiveness of a position according to all considerations of name</summary>
+		/// CtrlF
+		public int GetAttractiveness(WPos pos, Player firedBy, string considerationName)
+		{
+			var answer = 0;
+			var world = firedBy.World;
+			var targetTile = world.Map.CellContaining(pos);
+
+			if (!world.Map.Contains(targetTile))
+				return 0;
+
+			var namedConsiderations = GetConsiderationsOfName(considerationName);
+
+			foreach (var consideration in namedConsiderations)
+			{
+				var radiusToUse = new WDist(consideration.CheckRadius.Length);
+
+				var checkActors = world.FindActorsInCircle(pos, radiusToUse);
+				foreach (var scrutinized in checkActors)
+					answer += consideration.GetAttractiveness(scrutinized, firedBy.Stances[scrutinized.Owner], firedBy);
+			}
+
+			return answer;
+		}
+
+		/// <summary>Evaluates the attractiveness of a group of actors according to all considerations of the selected name</summary>
+		/// CtrlF
+		public int GetAttractiveness(IEnumerable<Actor> actors, Player firedBy, string considerationName)
+		{
+			var answer = 0;
+
+			var namedConsiderations = GetConsiderationsOfName(considerationName);
+
+			foreach (var consideration in namedConsiderations)
+				foreach (var scrutinized in actors)
+					answer += consideration.GetAttractiveness(scrutinized, firedBy.Stances[scrutinized.Owner], firedBy);
+
+			return answer;
+		}
+
 		public int GetNextScanTime(HackyAI ai) { return ai.Random.Next(MinimumScanTimeInterval, MaximumScanTimeInterval); }
 
 		/// <summary>Makes up part of a decision, describing how to evaluate a target.</summary>
 		public class Consideration
 		{
 			public enum DecisionMetric { Health, Value, None }
 
+			/// CtrlF - this was done to help with multistep support powers since each step would need its own target considerations
+			[Desc("Used to name the Consideration in the case of multistep support powers, Leave null or blank for code to ignore.")]
+			public readonly string ConsiderationName = null;
```

Comment:
Use ```String.Empty``` for this cases

Reply:
`null` should be fine as default. We use it in a lot of other places as well.

Reply:
Yeah, we're using IsNullOrEmpty where the string's used instead.

Reply:
Ya kinda spaced on what to use there so I'll make sure that gets fixed.

Reply:
use null